runners/google-cloud-dataflow-java/worker/windmill/src/main/proto/windmill.proto

```diff
@@ -146,6 +146,79 @@ message TagBag {
   optional int64 fetch_max_bytes = 6 [default = 0x7fffffffffffffff];
 }
 
+// For a given sharding key and state family, a TagMultimap is a collection of
+// `entry_name, bag of values` pairs, each pair is a TagMultimapEntry.
+//
+// The request_position, continuation_position and fetch_max_bytes fields in
+// TagMultimapEntry are used for the pagination and byte limiting of individual
+// entry fetch requests get(entry_name); while those fields in
+// TagMultimapFetchRequest and TagMultimapFetchResponse are used for full
+// multimap fetch requests entry_names() and entries().
+// Do not set both in a TagMultimapFetchRequest at the same time.
+message TagMultimapEntry {
+  optional bytes entry_name = 1;
+  // In update request: if true all values associated with this entry_name will
+  // be deleted. If new values are present they will be written.
+  optional bool delete_all = 2;
+  // In update request: The given values will be added to the collection and
+  // associated with entry_name.
+  // In fetch response: Values that are associated with this entry_name in the
+  // multimap.
+  repeated bytes values = 3;
+  // In fetch request: A previously returned continuation_position from an
+  // earlier read response. Indicates we wish to fetch the next page of values.
+  // If this is the first request, set to empty.
+  // In fetch response: copied from request.
+  optional int64 request_position = 4;
+  // In fetch response: Set when there are values after those returned above,
+  // but they were suppressed to respect the fetch_max_bytes limit. Subsequent
+  // requests should copy this to request_position to retrieve the next page of
+  // values.
+  optional int64 continuation_position = 5;
+  // In fetch request: Limits the size of the fetched values to this byte limit.
+  // A lower limit may be imposed by the service.
+  optional int64 fetch_max_bytes = 6 [default = 0x7fffffffffffffff];
+}
+
+message TagMultimapFetchRequest {
+  optional bytes tag = 1;
+  optional string state_family = 2;
+  // If true, values will be omitted in the response.
+  optional bool fetch_entry_names_only = 3;
+  // Limits the size of the fetched entries to this byte limit. A lower limit
+  // may be imposed by the service.
+  optional int64 fetch_max_bytes = 4 [default = 0x7fffffffffffffff];
+  // A previously returned continuation_position from an earlier fetch response.
+  // Indicates we wish to fetch the next page of entries. If this is the first
+  // request, set to empty.
+  optional bytes request_position = 5;
+  // Fetch the requested subset of entries only. Will fetch all entries if left
+  // empty. Entries in entries_to_fetch should only have the entry_name,
+  // request_position and fetch_max_bytes set.
+  repeated TagMultimapEntry entries_to_fetch = 6;
```

Comment:
what happens if fetch_max_bytes is set on the FetchRequest and the individual entries?